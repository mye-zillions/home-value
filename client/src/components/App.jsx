import React from 'react';
import ZestimateRow from './ZestimateRow.jsx';
import styled from 'styled-components';
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faAngleDown, faHome } from '@fortawesome/free-solid-svg-icons';
import fetchPropertyData from '../lib/get.js';
library.add(faAngleDown, faHome);

const Button = styled.button`
  background: white;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  font-weight: '900';
  font-size: 28px;
  line-height: 31px;
  color: #444444
  border-top: solid 1px #eee;
  border-bottom: solid 1px #eee;
  padding-top: 20px;
  padding-bottom: 20px;
  margin-bottom: 15px;
  :hover {
    
  }
  `;

const DropDownArrow = styled.span`
  size: 27px;
  line-height: 31px;
  color: #444444;
  text-align: right; 
  float: right;
  font-weight:;
`;
const HomeDetails = styled.div`
`;

const ZestimateHeaderContainer = styled.div`
  padding: 20px 0px 20px 0px;
`;
const ZestimateTitle = styled.div`
  float: center;
  text-align:center;
  style: normal;
  font-weight: 700;
  font-size: 20px;
  line-height: 30px;
  color: #444444;
`;

const ZestimateValue = styled.div`
  text-align:center;
  style: normal;
  font-weight: 700;
  font-size: 28px;
  line-height: 36px;
  color: #444444;
`;

const CollapsibleContent = styled.div`
  margin-top: 20px;
`;

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      propertyData: [{}],
      comparableHomesData: [{}],
      localHomesData: [{}], 
      photosData: [{}],
      isHidden: true
    };
  }

  componentDidMount () {
    fetch('http://localhost:3000/api/properties')
      .then((response) => {
        return response.json();
      })
      .then((data) => {
        this.setState({
          propertyData: data.propertyData, 
          comparableHomesData: data.comparableHomesData,
          localHomesData: data.localHomesData,
          photosData: data.photosData
        });
      });
  }
  
  toggleHidden () {
    this.setState({
      isHidden: !this.state.isHidden
    });
  }

  render () {
    console.log(this.state)
    return (

      <div> 
        <Button onClick={this.toggleHidden.bind(this)}>
          Home Value
          <DropDownArrow>
            <FontAwesomeIcon icon='angle-down' size='sm'/>
          </DropDownArrow>
        </Button>
        {!this.state.isHidden && 
          <Child 
            propertyData={this.state.propertyData} 
            comparableHomesData={this.state.comparableHomesData}
            localHomesData={this.state.localHomesData}
            photosData={this.state.photosData}
          />
        }  
      </div>
    );
  }
}

const Child = ({propertyData, comparableHomesData, localHomesData, photosData}) => {
  return (
    <div>
      <CollapsibleContent>
        <HomeDetails>
          <ZestimateHeaderContainer>
            <ZestimateTitle>
            Zestimate
            </ZestimateTitle>
            <ZestimateValue> 
              {`$${propertyData[0].zestimationPrice}`}
            </ZestimateValue>
          </ZestimateHeaderContainer>
        </HomeDetails>
      </CollapsibleContent>
    
      <ZestimateRow 
        propertyData={propertyData}
        comparableHomesData={comparableHomesData}
        localHomesData={localHomesData}
        photosData={photosData}
      /> 
    </div>
  );
};

export default App; 