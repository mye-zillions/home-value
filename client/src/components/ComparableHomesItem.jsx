import React from 'react';
import styled from 'styled-components';

const PropertyCard = styled.div`
  border-top: solid 1px #eee;
  padding-top: 20px;
  padding-bottom: 20px;
  white-space: nowrap;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  font-size: 15px;
  font-weight: 700;
  line-height: 1.5;
  color: #444;

  :after, :before {
    bow-sizing: border-box;
  }
`;

const PropertyDetailSection = styled.div``;


const HomeStatus = styled.h5``;

const RecentlySoldIcon = styled.span`
  font-size: 13px;
  line-height: 15px;
  font-weight: 700;
`;

const Price = styled.div`
  font-size: 28px;
  line-height: 40px;
  font-weight: 400;
`;

const PropertyDetailedInfo = styled.div`
  font-size: 15px;
  line-height: 20px;
  font-weight: 700;
`;

const PropertyAddress = styled.div`
  font-size: 15px;
  line-height: 20px;
  font-weight: 400;
  color: #0074e4;
  : hover {
    color: #62aef7;
  }
`;

const PropertySqftCalculation = styled.span`
  flex: 3;
  text-align: right
  padding-top: 30px;
  font-size: 20px;
  line-height: 15px;
  font-weight: 400;
`;

var ComparableHomesItem = ({homeData}) => {
  return (
    <div>
      <PropertyCard>
        <PropertyDetailSection>
          <HomeStatus>
            <RecentlySoldIcon>
              <div>
                SOLD ({homeData.sellDate})
              </div>
            </RecentlySoldIcon>
          </HomeStatus>
          <Price>
            {`$${homeData.sellPrice}`}
          </Price>
          <PropertyDetailedInfo>
            {`${homeData.beds} bds - ${homeData.baths} ba - 1555 sqft`}                          
          </PropertyDetailedInfo>
          <PropertyAddress>
            {homeData.streetAddress}
          </PropertyAddress>
        </PropertyDetailSection>
        <PropertySqftCalculation>
          {`$${homeData.priceSqft}`}
        </PropertySqftCalculation>
      </PropertyCard>
    </div>
  );
};


export default ComparableHomesItem; 