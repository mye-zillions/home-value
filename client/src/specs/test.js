
import React from 'react';
import Enzyme, { mount, render, shallow} from 'enzyme';
import {App, Button, DropDownArrow} from '../components/App.jsx';
// import Modal from '..//components/Modal.jsx';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import ComparableHomesItem from '../components/ComparableHomesItem.jsx';

Enzyme.configure({adapter: new Adapter});

// A setuup function, one way of implementing tests
var setup = () => {
  const props = {
  };
  const wrapper = shallow(<App />);
  return { wrapper, props };
};

// Shallow rendering 
describe('App component', () => {
  it('Should exist', () => {
    const { wrapper } = setup();
    expect(wrapper.exists()).toBe(true);
  });

  // Shallow rendering 
  test('Should successfully mount App', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.exists()).toBe(true);
  });

  // it('Should render without throwing an error', () => {
  //   expect(shallow(<App />).find('<style>').exists()).toBe(true);
  // });

  // Shallow rendering 
  it('Should render the app component correctly', () => {  
    const AppComponent = renderer.create(<App />).toJSON();
    expect(AppComponent).toMatchSnapshot();
  });

  // full DOM rendering
  test('should full DOM render', () => {
    const wrapper = mount(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  // Shallow rendering for styled components
  test('Should successfully mount Button Styled Component', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper.exists()).toBe(true);
  });
  test('Should successfully mount DropDownArrow Styled Component', () => {
    const wrapper = shallow(<DropDownArrow />);
    expect(wrapper.exists()).toBe(true);
  });

  // Snapshot test of styled components
  test('Should cotain a button styled component', () => {
    const tree = renderer.create('<Button />').toJSON();
    expect(tree).toMatchSnapshot();
  });

  // child component should be hidden upon rendering 
  test('Should not show child component upon rendering', () => {
    const wrapper = shallow(<App />);
    const isHidden = wrapper.state().isHidden;
    expect(isHidden).toEqual(true);
  });
  
  // imported components
  it('Should contain imported Components', () => {
    const wrapper = shallow(<App />);
    const ZestimateRow = renderer.create('<ZestimateRow />').toJSON();
    expect(ZestimateRow).toMatchSnapshot();
  });
  
  test('Should have properties in state', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.state('isHidden')).toEqual(true);
    expect(wrapper.state('propertyData')).toEqual([{}]);
    expect(wrapper.state('comparableHomesData')).toEqual([{}]);
    expect(wrapper.state('localHomesData')).toEqual([{}]);
  });
  // invoking lifecycle method. 
  it('Should call componentDidMount', () => {
    const mount = jest.spyOn(App.prototype, 'componentDidMount');
    const wrapper = shallow(<App />);
    expect(App.prototype.componentDidMount).toHaveBeenCalled();
    mount.mockRestore();
  });

  // should have 4 properties in state ???

  //should fetch data 

  // should render data 

  // should show child component when button is clicked

  // test('Should show child component upon toggleing isHidden', () => {
  //   const wrapper = shallow(<Child />);
  //   const isHidden = wrapper.state().isHidden;
  //   expect(isHidden).toEqual(false);
  // });
});