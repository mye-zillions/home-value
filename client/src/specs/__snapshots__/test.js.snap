// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Should contain imported Components 1`] = `"<ZestimateRow />"`;

exports[`App component Should cotain a button styled component 1`] = `"<Button />"`;

exports[`App component Should render the app component correctly 1`] = `
.c0 {
  background: white;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-family: 'Segoe UI',Tahoma,Geneva,Verdana,sans-serif;
  font-weight: '900';
  font-size: 28px;
  line-height: 31px;
  color: #444444 border-top:solid 1px #eee;
  border-bottom: solid 1px #eee;
  padding-top: 20px;
  padding-bottom: 20px;
  margin-bottom: 15px;
}

.c1 {
  size: 27px;
  line-height: 31px;
  color: #444444;
  text-align: right;
  float: right;
}

<div>
  <button
    className="c0"
    onClick={[Function]}
  >
    Home Value
    <span
      className="c1"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-down fa-w-10 fa-sm "
        data-icon="angle-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`App component should full DOM render 1`] = `ReactWrapper {}`;
